core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -196,7 +198,37 @@ public RequestTemplate create(Object[] argv) {
           }
         }
       }
-      return resolve(argv, mutable, varBuilder);
+
+      RequestTemplate template = resolve(argv, mutable, varBuilder);
+
+      // add query map parameters after initial resolve so that they take
+      // precedence over any predefined values
+      return addQueryMapQueryParameters(argv, template);
+    }
+
+    private RequestTemplate addQueryMapQueryParameters(Object[] argv, RequestTemplate mutable) {
+    	for (int currQueryMapIndex : metadata.queryMapIndices()) {
```

Comment:
single index will allow you to early exit.. ex. `if (metadata.queryMapIndex == null) return`